File: examples/spring-ai-engram-cloud-demo/src/main/java/dev/jamjet/demo/springaiengram/ChatController.java
package dev.jamjet.demo.springaiengram;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ChatController {

    private final MemoryAgent agent;

    public ChatController(MemoryAgent agent) {
        this.agent = agent;
    }

    @PostMapping("/chat")
    public ChatResponse chat(@RequestParam String session, @RequestBody String message) {
        String reply = agent.chat(session, message);
        return new ChatResponse(session, reply);
    }

    public record ChatResponse(String session, String reply) {}
}

File: examples/kotlin-koog-engram-cloud-demo/src/main/kotlin/dev/jamjet/demo/koogengram/ChatController.kt
package dev.jamjet.demo.koogengram

import kotlinx.coroutines.runBlocking
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class ChatController(private val agent: MemoryAgent) {

    @PostMapping("/chat")
    fun chat(@RequestParam session: String, @RequestBody message: String): ChatResponse =
        runBlocking {
            val reply = agent.chat(session, message)
            ChatResponse(session = session, reply = reply)
        }

    data class ChatResponse(val session: String, val reply: String)
}
